ts/components/conversation/InviteFriendsDialog.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { Contact, MemberList } from './MemberList';
+
+interface Props {
+  friendList: Array<any>;
+  chatName: string;
+  onSubmit: any;
+  onClose: any;
+}
+
+declare global {
+  interface Window {
+    i18n: any;
+  }
+}
+
+interface State {
+  friendList: Array<Contact>;
+}
+
+export class InviteFriendsDialog extends React.Component<Props, State> {
```

Comment:
Would you say it makes sense to move the logic into the InviteFriendsDialogView (backbone layer) and keep the react component as dumb as possible by only taking care of rendering?

Reply:
I'm not sure. I prefer to keep the backbone part minimal since we seem to be moving away from it. It is not that uncommon to have logic in a React component either.

Reply:
Sounds reasonable !